architecture-decisions/0002-indexing.md

```diff
@@ -0,0 +1,86 @@
+# 2. Indexing Architecture
+
+Date: 2024-07-09
+
+## Status
+
+Accepted
+
+## Context
+
+DPUL-Collections must have a resilient indexing pipeline that can quickly harvest, transform, and index millions of records. We foresee needing to index millions of records, regularly change weighting algorithms, and accept records from external institutions which may not be stable in the long term.
+
+There must be a verifiable method of ensuring that 100% of Figgy's relevant records are indexed into DPUL-Collections, to prevent us from constantly scrambling and diagnosing indexing issues as we do now with our spotlight-powered DPUL.
+
+We will be starting with indexing from Figgy, so that's where our initial performance requirements will be based upon.
+
+Often times systems like this use event streaming platforms such as Kafka, but we'd like to prevent adding new technology to our stack. We think we can use Postgres tables as a compact event log.
```

Comment:
Perhaps add that we rely on figgy's single-node database architecture for the integrity of our event log